pkg/build/api/types.go

```diff
@@ -199,6 +199,9 @@ type STIBuildStrategy struct {
 	// Image is the image used to execute the build.
 	Image string `json:"image,omitempty"`
 
+	// From is reference to an image repository from where the docker image should be pulled
+	From kapi.ObjectReference `json:"from,omitempty"`
```

Comment:
I will make it a pointer, Docker and Custom builds get From field. There's a lot of refactoring involved.